### DISCUSSION TITLE
Object Animation in DeckGL

### DISCUSSION THREAD

**Discussion by hoang-ta** [visgl/deck.gl#6118] [General]
Hello, I am trying to animate an objects from layers ( **GeoJson layer, icon layer, etc.**) of DeckGL like in this gif: 
![animate](https://user-images.githubusercontent.com/61722520/130008318-2c28ab04-7a70-4123-8db6-dcc6a8586ffe.gif)

Do you know where should I start with? I have read through docs but still not understand yet.

1. https://deck.gl/docs/developer-guide/animations-and-transitions
*  Does the above animation relate to property animation ?
2. https://deck.gl/examples/trips-layer/
* According to my current understanding, only **trips layer** supports animation, doesn't it ?
3. https://deck.gl/docs/developer-guide/custom-layers/writing-shaders#animation-vertex-shader
* Does it imply here that **vertex shader** is used for the type of animation above ?

**Comment by Pessimistress** [comment]
See [props.transitions](https://deck.gl/docs/api-reference/core/layer#transitions)

**Reply by hoang-ta** [reply]
I have tried your suggestion and have a problem for my demo which is: 

- Moving a red rectangle (path layer) on a line with points (geojson layer). The rectangle starts moving from point (0,0) then I provide a callback to increase x by 10 at the end of the transition (`onEnd`) until the rectangle reachs point (30,0).
- The animation works, but there are still a problem: **the retangle is rotated at the end of each transition**

![transition](https://user-images.githubusercontent.com/61722520/133045264-1940951c-1d39-4d91-839d-953ec8e87900.gif)


[Sandbox code](https://codesandbox.io/s/map-transition-k30tn?file=/src/App.tsx)

**Reply by hoang-ta** [reply]
Do you have any suggestion @Pessimistress ? Currently, I am looking at documents related to shader and an example of subclass a layer like this one: https://observablehq.com/@pessimistress/deck-gl-tutorial-subclassing-a-layer
